consensus/obcpbft/obc-batch.go

```diff
@@ -310,6 +316,11 @@ func (op *obcBatch) processMessage(ocMsg *pb.Message, senderHandle *pb.PeerID) e
 	}
 
 	if req := batchMsg.GetRequest(); req != nil {
+		if !op.deduplicator.IsNew(req) {
```

Comment:
I think this is okay here.  Assuming the non-byzantine case (this patch acknowledges that it enables censorship in the byzantine case), there are really only two scenarios.

1. We have executed the request, or a later request.  In which case it does not make it into our outstanding requests, and we discard it.
2. We have not executed the request or any later request.  In this case, we will add it to our outstanding requests.  If the request has already prepared, this is not a problem, as if we are the primary, it is already in our list of `pendingRequests`, so we will not order it a second time.  If we are not the primary, then we will build the `pendingRequests` list after the view change, also ensuring we do not doubly execute.